ValidationMixin.js

```diff
@@ -1,83 +1,53 @@
-"use strict";
+'use strict';
 
-var assign = Object.assign || require('object.assign');
+require('object.assign').shim();
+var result = require('lodash.result');
 var ValidationFactory = require('./ValidationFactory');
 
 var ValidationMixin = {
-
   /**
-   * Validate single form field against the components state. If no field
-   * is provided, validate entire form.
-   *
-   * @param {?String} field State key to validate
-   * @return {Object} newly updated errors object keyed on state field
-   * names. Missing key or undefined value indicates no error.
+   * Check for sane configurations
    */
-  validate: function(field) {
-    var validatorTypes = this.validatorTypes || {};
-    if (typeof this.validatorTypes === 'function') {
-      validatorTypes = this.validatorTypes();
+  componentDidMount: function() {
+    if (typeof this.validatorTypes !== 'function' && !Array.isArray(this.validatorTypes)) {
+      throw Error('invalid `validatorTypes` type');
```

Comment:
should we also allow validatorTypes to be undefined? I think it's more user friendly to not blow up with the addition of a mixin. Thoughts?